pallets/multisig/src/lib.rs

```diff
@@ -735,79 +853,165 @@ impl<T: Trait> Module<T> {
-                    .dispatch(frame_system::RawOrigin::Signed(multisig.clone()).into())
-                {
-                    Ok(_) => true,
+            let res =
+                match proposal.dispatch(frame_system::RawOrigin::Signed(multisig.clone()).into()) {
+                    Ok(_) => {
+                        proposal_details.status = ProposalStatus::ExecutionSuccessful;
+                        true
+                    }
                     Err(e) => {
                         let e: DispatchError = e;
                         sp_runtime::print(e);
+                        proposal_details.status = ProposalStatus::ExecutionFailed;
                         false
                     }
                 };
-                let current_did = Context::current_identity::<Identity<T>>()
-                    .ok_or_else(|| Error::<T>::MissingCurrentIdentity)?;
-                Self::deposit_event(RawEvent::ProposalExecuted(
-                    current_did,
-                    multisig,
-                    proposal_id,
-                    res,
-                ));
-                Ok(())
-            } else {
-                Ok(())
+            Self::deposit_event(RawEvent::ProposalExecuted(
+                current_did,
+                multisig,
+                proposal_id,
+                res,
+            ));
+        }
+        Ok(())
+    }
+
+    /// Rejects a multisig proposal
+    fn unsafe_reject(
+        multisig: T::AccountId,
+        signer: Signatory,
+        proposal_id: u64,
+    ) -> DispatchResult {
+        ensure!(
+            <MultiSigSigners<T>>::contains_key(&multisig, &signer),
+            Error::<T>::NotASigner
+        );
+        let multisig_signer_proposal = (multisig.clone(), signer, proposal_id);
+        let multisig_proposal = (multisig.clone(), proposal_id);
+        ensure!(
+            !Self::votes(&multisig_signer_proposal),
+            Error::<T>::AlreadyVoted
+        );
+        let mut proposal_details = Self::proposal_detail(&multisig_proposal);
+        proposal_details.rejections += 1u64;
+        let current_did = Context::current_identity::<Identity<T>>().unwrap_or_default();
+        match proposal_details.status {
+            ProposalStatus::Invalid => return Err(Error::<T>::ProposalMissing.into()),
+            ProposalStatus::Rejected => return Err(Error::<T>::ProposalAlreadyRejected.into()),
+            ProposalStatus::ExecutionSuccessful | ProposalStatus::ExecutionFailed => {
+                return Err(Error::<T>::ProposalAlreadyExecuted.into())
+            }
+            ProposalStatus::ActiveOrExpired => {
+                // Ensure proposal is not expired
+                if let Some(expiry) = proposal_details.expiry {
+                    ensure!(
+                        expiry > <pallet_timestamp::Module<T>>::get(),
+                        Error::<T>::ProposalExpired
+                    );
+                }
+                if proposal_details.auto_close {
```

Comment:
If `auto_close` is false then do we really need to allow the participants to reject the proposal? It will be worthless to do as they can show their rejection by not voting on that proposal.

Reply:
Since changing your votes is not allowed, explicitly rejecting a proposal has value even if auto_close is disabled.

Reply:
how does it reflect the value even when changing of the votes is not allowed? It doesn't affect the proposal in any way. It will give the same behaviour even if they did not vote for the proposal (It will save the gas as well).

Reply:
Casting a reject vote proves that the user will not be voting in favor of this proposal. Casting the reject vote is similar to "Abstain" in this case. If the user has not cast any vote, they still have a chance to vote positively and have the proposal executed. If the cast reject vote, they can not cast an approve vote later.

Reply:
This kind of approach can be applied for the cooperate voting where some proposals need 100% voting attendance but for Multisig IMO it is kind of overburden. I am agnostic about the approach though @adamdossa WDYT?

Reply:
I agree that a -ve vote does add signal (that you can't support the proposal in future) so seems reasonable, even if it doesn't necessarily impact the outcome of the vote directly.